app/models/unique_name.rb

```diff
@@ -5,17 +5,23 @@
 ###
 
 class UniqueName < ApplicationRecord
-
   def self.update!
     Rails.logger.info 'Updating the unique names table'
 
     # Build double metaphone representations for all names in the database
-    names = GrdaWarehouse::Hud::Client.source.select('FirstName').distinct.pluck('FirstName')&.map{ |n| n&.downcase || ''} + GrdaWarehouse::Hud::Client.source.select('LastName').distinct.pluck('LastName')&.map{ |n| n&.downcase || ''}
-    names.uniq.each do |name|
+    existing_names = UniqueName.pluck(:name)
```

Comment:
I wonder. Is the name property actually constrained to be unique in the database? Not that it matters here but curious about that

Reply:
Looks like not currently.